### PR TITLE
fix(iroh-net): Ensure netcheck finishes once it has results

### LINKED ISSUE #2021: netcheck report does not finish early enough
I have a netcheck report where we have established connectivity to the derper and figured out our IPv4 STUN addresses very quickly, yet keep waiting for all probes much, much to long.  In fact it logs `Have enough probe reports, aborting further probes soon` and yet it waits forever for to actually abort.  This wreaks havoc with good connection establishment.

Splitting this off to keep #1984 somewhat limited in scope.

Relevant log: https://gist.github.com/flub/1d503f757bc1e275e417d0d918b66604

### PR DESCRIPTION
**Pull request by flub**
## Description

Once netcheck has managed to establish an external IP address there is
no need to further wait on the portmapper.  The results are already
useful.  If later the portmapper manages to get something done the
magicsock will react to that and make use of it.

## Notes & open questions

Fixes #2021

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]
code looks good, what impact have you seen in testing?

File: iroh-net/src/netcheck/reportgen.rs [hunk 1]
```diff
@@ -170,6 +170,12 @@ struct Actor {
     last_report: Option<Arc<Report>>,
     /// The portmapper client, if there is one.
     port_mapper: Option<portmapper::Client>,
+    /// Whether the actor should try to reach things other than localhost.
+    ///
+    /// This is set to true in tests to avoid probing the local LAN's router etc.
+    ///
+    /// TODO: currently this only skips the portmapper, but still does STUN, HTTP and ICMP
+    /// probes.  Furthermore none of our tests actually use this feature.
```
Comment (dignifiedquire) [inline iroh-net/src/netcheck/reportgen.rs:178]: honestly, just delete it, we can bring it back if we ever need to

**Comment by flub** [comment]
Pinging @divagant-martian to check my understanding of the portmapper is correct.

I think what happens is that on the very first netcheck the portmapper has not yet run to completion and that slows down the first netcheck, often only to not get a portmapper result anyway.  Later on there are cached reports that are returned.  As all this is reporting is a bunch of bools in an option I think it's fine to not wait for this the first time.

File: iroh-net/src/netcheck/reportgen.rs (hunk 1 above)
Comment (flub) [reply iroh-net/src/netcheck/reportgen.rs:178]: yeah, fair enough.  i'll do that in a separate PR though.

**Comment by flub** [comment]
@divagant-martian I'm going to interpret the thumbs-up as approval.  Can always revert if that was wrong.

**Pull request merged**